### PR TITLE
Added iCal support

### PR DESCRIPTION
**Pull request by kjoconnor**
Added iCal support as per issue http://code.google.com/p/sickbeard/issues/detail?id=389

It makes use of PIL if it's available and needs the iCalendar library - http://pypi.python.org/pypi/icalendar/2.1 .  The library may need a patch (https://github.com/cozi/icalendar/pull/1) to work properly with Google Calendar though.

Tested on iPhone & Google Calendar.

### CODE REVIEW

**Comment by tester22** [comment]
Why not add the module to the library folder in SB, that reduces the need for the user to install modules to make it work "out of the box".

**Comment by bjeanes** [comment]
I'd love to see this!

**Comment by camflan** [comment]
Was going to do this myself, nice work. Pull this soon :)

**Comment by smidley** [comment]
+1

**Comment by kjoconnor** [comment]
I've resubmitted it twice already with no input from @midgetspy as to
why it's not being accepted. If I can get some guidance as to what's
wrong with it I'll fix and resubmit again.

On Jun 27, 2012, at 4:57, aanderson
reply@reply.github.com
wrote:

**Comment by ghost** [comment]
+1 

Would love to see this added

File: sickbeard/webserve.py [hunk 1]
```diff
@@ -2549,6 +2549,117 @@ def comingEpisodes(self, layout="None"):
 
         return _munge(t)
 
+    # Provides a URL to generate an .ics/iCalendar subscribable URL to have an upcoming episode schedule
+    @cherrypy.expose
+    def ical(self):
```
Comment (midgetspy) [inline sickbeard/webserve.py:2554]: The indentation in this function looks very screwy in this diff, is there a problem with tabs vs spaces?

File: sickbeard/webserve.py [hunk 2]
```diff
@@ -2549,6 +2549,117 @@ def comingEpisodes(self, layout="None"):
...
+            	event['uid'] = datetime.date.today().isoformat() + "-" + str(random.randint(10000,99999)) + "@Sick-Beard"
+            	
+            	# Add the event to the Calendar object
+            	cal.add_component(event)
+        
+        # Pass the calendar string to the tmpl file
+        t = PageTemplate(file="ical.tmpl")
+        t.calendarString = cal.as_string()
+        
+        return _munge(t)
```
Comment (midgetspy) [inline sickbeard/webserve.py:2661]: Why not just return a string?

**Comment by kjoconnor** [comment]
@midgetspy I redid this in #464

**Pull request closed without merging**